### PR TITLE
deps: migrate to `@electron/notarize`

### LINKED ISSUE #3669: Migrate notary tool
### What should we do?

Migrate to `@electron/notarize`.

### Why should we do it?

The notarization package we use right now is deprecated, and it uses a MacOS utility tool which goes out of support on 1st November 2023.

### PR DESCRIPTION
**Pull request by barmac**
Closes #3669

### CODE REVIEW

**Comment by barmac** [comment]
I am verifying it works at https://github.com/camunda/camunda-modeler/actions/runs/5265891992/jobs/9519124382

**Comment by barmac** [comment]
Second try: https://github.com/camunda/camunda-modeler/actions/runs/5266399516/jobs/9520272611

File: package.json [hunk 1]
```diff
@@ -89,5 +88,8 @@
         false
       ]
     ]
+  },
+  "dependencies": {
+    "@electron/notarize": "^1.2.3"
```
Comment (marstamm) [inline package.json:93]: Does this need to be a production dependency?

Comment (barmac) [reply package.json]: Nope!

**Pull request merged**